pkg/controller/resourcedistribution/resourcedistribution_namespace_event_handler.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcedistribution
+
+import (
+	"context"
+	"reflect"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.EventHandler = &enqueueRequestForNamespace{}
+
+type matchFunc func(*corev1.Namespace, *appsv1alpha1.ResourceDistribution) (bool, error)
+
+type enqueueRequestForNamespace struct {
+	reader client.Reader
+}
+
+func (p *enqueueRequestForNamespace) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	p.addNamespace(q, evt.Object, matchViaTargets)
+}
+func (p *enqueueRequestForNamespace) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	p.addNamespace(q, evt.Object, matchViaIncludedNamespaces)
```

Comment:
Why can not use `matchViaTargets ` here?

Reply:
We watch namespace delete event just for updating conditions (status.conditions[NoUncreatedNamespaceID]), and this condition  needs to be updated only when this namespace in IncludedNamespaces.